python/ray/experimental/state.py

```diff
@@ -594,31 +626,141 @@ def micros_rel(ts):
+                        "tid": parent_info["worker_id"],
+                        "ts": micros_rel(task_profiles[task_table[task_id]
+                                                       ["TaskSpec"]
+                                                       ["ParentTaskID"]]
+                                          ["get_arguments_start"]),
+                        "ph": "s",
+                        "name": "SubmitTask",
+                        "args": {},
+                        "id": str(worker) + str(micros(min(parent_times)))
+                    }
+                    full_trace.append(parent)
+
+                    task_trace = {
+                         "cat": "submit_task",
+                         "pid": "Node " + str(worker["node_ip_address"]),
+                         "tid": info["worker_id"],
+                         "ts": micros_rel(info["get_arguments_start"]),
+                         "ph": "f",
+                         "name": "SubmitTask",
+                         "args": {},
+                         "id": str(worker) + str(micros(min(parent_times))),
+                         "bp": "e"
+                    }
+                    full_trace.append(task_trace)
+
+            if obj_dep:
+                args = task_table[task_id]["TaskSpec"]["Args"]
+                for arg in args:
+                    if isinstance(arg, ray.local_scheduler.ObjectID):
+                        continue
+                    object_info = self._object_table(arg)
+                    if object_info["IsPut"]:
+                        continue
+                    if arg not in seen_obj:
+                        seen_obj[arg] = 0
+                    seen_obj[arg] += 1
+                    owner_task = self._object_table(arg)["TaskID"]
+                    owner_worker = (workers[task_profiles
+                                    [owner_task]["worker_id"]])
+                    owner = {
+                        "cat": "obj_dependency",
+                        "pid": "Node " + str(owner_worker["node_ip_address"]),
+                        "tid": task_profiles[owner_task]["worker_id"],
+                        "ts": micros_rel(task_profiles[owner_task]
+                                         ["store_outputs_end"]) - 2,
+                        "ph": "s",
+                        "name": "ObjectDependency",
+                        "args": {},
+                        "bp": "e",
+                        "cname": "cq_build_attempt_failed",
+                        "id": str("obj") + str(arg) + str(seen_obj[arg])
+                    }
+                    full_trace.append(owner)
+
+                    dependent = {
+                        "cat": "obj_dependency",
+                        "pid":  "Node " + str(worker["node_ip_address"]),
+                        "tid": info["worker_id"],
+                        "ts": micros_rel(info["get_arguments_start"]) + 2,
```

Comment:
Is this `+2` the thing you needed to do to get the arrows to show up reliably? Can you add a comment with an in-depth explanation of the problem and the solution? I still don't understand it, so it'll be useful for people looking at this code in the future.